devkit/states/vim/pr_vimrc_init

```diff
@@ -1,22 +1,364 @@
-" Set :GoDef and :GoInfo to use gopls
+" get all of the plugins
+source ~/.pr_vimrc_plug
+
+"-----------------------
+" airblade/vim-gitgutter
+"-----------------------
+" TODO disables default key maps why
+let g:gitgutter_map_keys = 0
+let g:gitgutter_max_signs = 1000
+let g:gitgutter_realtime = 0
+let g:gitgutter_eager = 0
+let g:gitgutter_diff_args = '--ignore-space-at-eol'
+nnoremap <S-F5> :GitGutterToggle<CR>
+nnoremap <S-F6> :GitGutterSignsToggle<CR>
+nnoremap <S-F7> :GitGutterLineHighlightsToggle<CR>
+nmap <silent> ]h :GitGutterNextHunk<CR>
+nmap <silent> [h :GitGutterPrevHunk<CR>
+nnoremap <silent> <leader>ggr :GitGutterRevertHunk<CR>
+nnoremap <silent> <leader>ggp :GitGutterPreviewHunk<CR>
+nnoremap <silent> <leader>gga :GitGutterAll<CR>
+
+"------------------
+" ervandew/supertab
+"------------------
+let g:SuperTabDefaultCompletionType = "context"
+let g:SuperTabContextTextOmniPrecedence = ['&omnifunc', '&completefunc']
+
+"-------------
+" fatih/vim-go
+"-------------
+" Default: auto-detect a snippet engine
+let g:go_snippet_engine = "ultisnips"
+
+" Default: gofmt; using goimports extends functionality
+let g:go_fmt_command = "goimports"
+" Default: show a location list when gofmt fails
+let g:go_fmt_fail_silently = 1
+
+" Default: empty; set to quickfix so there is one method for navigation
+let g:go_list_type = "quickfix"
+
+" Default: guru; use gopls for :GoDef
 let g:go_def_mode='gopls'
+" Default: gocode; use gopls for :GoInfo
 let g:go_info_mode='gopls'
+" Default: show gopls won't work outside GOPATH warning
+let g:go_null_module_warning = 0
+" Default: work with newly modified GOPATH; detect before using gopls
+let g:go_autodetect_gopath = 1
+
+" Default: set :GoTest timeout to 10 seconds
+let g:go_test_timeout = '10s'
+" Default: do not prepend name of test to error message
+let g:go_test_prepend_name = 1
+
+" Default: highlight all instances of identifier under cursor
+" let g:go_auto_sameids = 0
+" Default: do not show :GoInfo for word under cursor
```

Comment:
I enable this by default, doesn't seem to cause a perf  problem

Reply:
done